lib/cloudant.js

```diff
@@ -524,21 +531,52 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var idName = self.idName(model);
+  // if idName exists, change the idName to be the default `_id` key
+  if (idName && data[idName]) {
+    data._id = data[idName];
+    delete data[idName];
+  }
+  var res = {};
+  var docList = [];
+
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
     if (docs.length === 0) return cb(null, {count: 0});
 
     data = getPlainJSONData.call(self, model, data);
+
+    /**
+     * for each of the docs from the given filter, only update the instance where the revs match
+     * behavior of bulk update in cloudant:
+       - if both _id and _rev are not provided, return empty list
+       - if _id is provided and _rev is not provided, throw error
+       - if _id is not provided and _rev is provided, do update
+       - if both _id and _rev are provided, do update
+    **/
     async.each(docs, function(doc, cb) {
-      _.mergeWith(doc, data, function(dest, src) { return src; });
+      if (doc._id === data._id || doc._rev === data._rev) {
+        delete doc._id;
```

Comment:
Not necessary, and frankly, not correct.
We want to make sure that what we're merging is being merged correctly.